### ISSUE TITLE
tracks will not play

### ISSUE THREAD

**Issue by kiingkiller** [Soundnode/soundnode-app#802]
it was working fine this morning but when i got home when ever i try and play a track nothing happends and if i try and click into the track i get the oh no message.

**Comment by wb9688** [comment]
Try removing the LocalStorage in the Soundnode setting and then rrstart
Soundnode…

**Comment by Flukky** [comment]
Problem appeared 30.06.2016. It is not possible to listen any track. Everything I got it is silence. When I try to move forward a current track, I got the message: "Oh No. Something went wrong. I can't play this track :(". I tried to clean up local storage, run as admin. It didn't help.

**Comment by eisenbergrobin** [comment]
Same here :( Tried removing the LocalStorage, no help. Version 0.6.4.

**Comment by clemlycao** [comment]
Same problem i clean the LocalStorage and download the app again from the website. It's work now :)

**Comment by clivend** [comment]
Same here :( Tried removing the LocalStorage, no help. Version 0.6.4 on Ubuntu 14.04

**Comment by Mattijah** [comment]
Exactly same problem here. Version `0.6.4`, appeared yesterday.

**Comment by jakejarrett** [comment]
It appears that we exceeded the [Rate limit](https://developers.soundcloud.com/docs/api/rate-limits)

```
stream:1 GET https://api.soundcloud.com/tracks/XX/stream?client_id=XX
429 (Too Many Requests)
```

**Comment by GrayedFox** [comment]
@jakejarrett hey there, how did you find that log message? I haven't built from source (should probably do that) - but since it's a nodejs app - can I enable logging somehow? view a console even? would be good to know so I can make more meaningful bug reports!

also +1 on this issue. Clearing local cache and restarting app resolved for me.

**Comment by jakejarrett** [comment]
@GrayedFox You can just press Control (Command on Mac) + / and it will open the devtools :smile:

**Comment by GrayedFox** [comment]
And on Linux? :penguin:

**Comment by jakejarrett** [comment]
Linux is Control + / :smile_cat:

**Comment by wb9688** [comment]
@jakejarrett how could we avoid exceeding the rate limit? I was thinking of
at least implementing session-based caching (so when you listen again to a
song you won't make a new request) as allowed in the [Terms of Use](
https://developers.soundcloud.com/docs/api/terms-of-use#caching):

> Your app may employ session-based caching, but only to the extent
reasonably necessary for the operation of your app during that session, and
any cached content must cease to be available, accessible or playable
within your app at the end of that session.

**Comment by weblancaster** [comment]
We hit the Soundclou API limit that's the reason we can't play tracks.

**Comment by wb9688** [comment]
@weblancaster Jake Jarrett has already said that, what about my idea above
to hit the rate limit less fast?

**Comment by Pitros** [comment]
Cache doesn't sound good for me, especially that most queries comes from playing stream, so cache would let us save maybe few percents at most.

**Comment by wb9688** [comment]
And what about adding an option for using the user's Client ID?

**Comment by GrayedFox** [comment]
Eh? Cache would potentially save the end user a lot of data usage, and reduce how many times we hit the API. Not sure the best way to get the stats, but I could record all my traffic coming from Soundnode over several sessions using a proxy and gather data based on those requests most repeated. This would give us a base to decide what to cache first, based on traffic. Would be more useful with a larger user base (I.e. not just me) but hey, it's a start :)

**Comment by eisenbergrobin** [comment]
I think caching will barely (if at all) help with this issue. 

How about using a rotating pool of clientIds/tokens? If we just generate a few tokens and rotate between them wouldn't that mitigate the problem?

**Comment by Pitros** [comment]
When soundcloud added limits, we talked about that but I don't remember what @weblancaster thought when I proposed using more api tokens and just randomly assigning them while user logs to app. That is still the best idea for me.

**Comment by weblancaster** [comment]
I'm just going to build something to let me know when Soundnode is close to hit the limit and change to another developer id (I'm going to create two more developer ID)

**Comment by weblancaster** [comment]
#804 and #805

**Comment by GrayedFox** [comment]
@weblancaster that's not a scalable solution at all my friend. Also, possibly breaks terms of service?

Why don't we just ask the kids at Soundcloud what they think is the best solution? Maybe they have workarounds

**Comment by weblancaster** [comment]
You are correct not scalable but that's the only thing we can do for a short term (I also have asked someone at Soundcloud.. there is nothing we can do)

**Comment by onedrawingperday** [comment]
Clearing local storage and restarting Soundnode does not work for me.

Here are the contents of the JSON

> {"errors":[{"meta":{"rate_limit":{"bucket":"by-client","max_nr_of_requests":65000,"time_window":"PT24H","name":"plays"},"remaining_requests":0,"reset_time":"2016/08/30 05:08:57 +0000"}}]}

Maybe there should be an option in the settings for the end-user to enter a developer id, if they wish.

I'm on Soundnode 0.6.4, since 0.6.5 is broken on Ubuntu 16.04 and I am in no mood to compile it from source (have better things to do with my time).

**Comment by bryguy** [comment]
@onedrawingperday While not as friendly as a field in settings, you can open up `Help > Developer Tools > Console`, then run the following command:

```
window.localStorage.scClientId = "YOUR_PERSONAL_SOUNDCLOUD_CLIENT_ID"; // replace with your actual SoundCloud Client ID
```

It should persist across app relaunches so that you can keep rocking out even in the rare event that Soundnode's client id has been rate limited. See #858.

Hope that helps if this hasn't been resolved more directly by an app update before the next time the rate limit is exceeded.

**Comment by onedrawingperday** [comment]
Thanks @bryguy will check this out as soon as time permits.

**Comment by poltergeistG** [comment]
resetting localstorage did'nt solve the issue for me, nor did re downloading, help would be apreciated :)

**Comment by Racer500** [comment]
It seems we hit the rate limit again...

**Comment by weblancaster** [comment]
yes we did @Racer500 .. I'm going to figure out something about it.

**Comment by poltergeistG** [comment]
any updates? i run a really weak computer and the only way for me to listen to music and multi - task w/o google taking all my resources is with this :)

**Comment by weblancaster** [comment]
@poltergeistG nothing yet.. I have contacted Soundcloud to ask for more bandwidth but they never replied to me.

**Comment by bryguy** [comment]
@poltergeistG 

I was going to provide step-by-step instructions for setting up a soundcloud app and using that as your personal client id for Soundnode, but apparently that's become far more difficult process fairly recently:

![screen shot 2016-11-04 at 4 25 09 pm](https://cloud.githubusercontent.com/assets/1933777/20026258/7b0fbfca-a2b5-11e6-9cf4-a198c73300bc.png)

**Comment by phoen1x74** [comment]
@bryguy where is that menu? Late comment, but still. Would like to know a fix for this.

**Comment by bryguy** [comment]
@phoen1x74 

On OS X:

![os x soundnode devtools](https://cloud.githubusercontent.com/assets/1933777/20124690/2b12ab92-a5dd-11e6-9e39-b97d8ccf3c17.png)
![os x soundnode clientid](https://cloud.githubusercontent.com/assets/1933777/20124691/2e990d2e-a5dd-11e6-87e1-1d42f02d3862.png)

**Comment by phoen1x74** [comment]
@bryguy Yeah, looks like on the electron-build devtools isn't present for some reason. Guess I'll have to downgrade to 0.64 as 65 is broken for me.

**Comment by bryguy** [comment]
@phoen1x74 Interesting, I'm running version 0.6.5.

**Comment by weblancaster** [comment]
@phoen1x74 download the latest version here and you will be able to see the devtools https://github.com/Soundnode/soundnode-app/pull/894

**Comment by phoen1x74** [comment]
Thanks @weblancaster

**Comment by phoen1x74** [comment]
Mind compiling it for me @weblancaster; grunt seems to be buggy for me. I get an error when trying to build.

**Comment by weblancaster** [comment]
All unfortunately Soundcloud made harder to create more apps making impossible for me to request more tokens to swap when the limit is reached.

any ideas besides each user create a new app and use the token on Soundnode?

**Comment by phoen1x74** [comment]
@weblancaster When compiling with `grunt build`, I get an error `>> Error: Cannot find module 'load-grunt-config'
`. I can continue with --force, but it doesn't compile successfully. Latest build.

**Comment by weblancaster** [comment]
@phoen1x74  did you download from the dropbox or pull locally the PR from #894 

if you pull locally, remove the node_modules folder then install again `npm i` then you can use `NODE_ENV=development npm start` to start Soundnode on Electron

**Comment by phoen1x74** [comment]
Alright, so that worked to compile it, but now it is stuck on "Authenticating your Soundcloud Account," and not doing anything.

**Comment by weblancaster** [comment]
@phoen1x74 yeah that's the part that I'm working on trying to figure out since electron windows works differently than nw.js

**Comment by Danco1** [comment]
I'm having the same issue from a month ago, i deletes my localstorage and reinstalled the app, but having the same response on my console "Failed to load resource: the server responded with a status of 429 (Too Many Requests)". There's a short term solution for this issue while the Devs solve it?

**Comment by wb9688** [comment]
@weblancaster: I found a way to get around the API limit: just use SoundCloud's `client_id`. I've made a quick Python implementation of it, but that only works with username/password auth and if a captcha isn't required:

```py
import getpass
import json
import re
import requests

client_id = re.compile(',client_id:"(.*?)"').search(requests.get(re.compile('src="https://a-v2.sndcdn.com/assets/app-.*.js"').search(requests.get('https://soundcloud.com/').text).group()[5:-1]).text).group()[12:-1]

username = raw_input('Username: ')
password = getpass.getpass()

access_token = requests.post('https://sign-in.soundcloud.com/sign-in/password?client_id={}&app_version=1479467323'.format(client_id), data=json.dumps({'client_id': client_id, 'scope': 'fast-connect non-expiring purchase signup upload', 'recaptcha_response': '', 'credentials': {'identifier': username, 'password': password}}), headers={'Content-type': 'application/json'}).json()['session']['access_token']

print requests.get('https://api-v2.soundcloud.com/me?client_id={}&app_version=1479467323'.format(client_id), headers={'Authorization': 'OAuth {}'.format(access_token)}).json()
```

**Comment by Legitgamer264** [comment]
@wb9688 I am getting "Import requests, no such module named requests"

**Comment by wb9688** [comment]
@Legitgamer264: Then you haven't installed the commonly used `requests`
Python module, you can install it with `pip install requests` (might need
`sudo`) or your distro's package manager if you're using Linux (eg. `sudo
apt install python-requests`)…

**Comment by Legitgamer264** [comment]
@wb9688 Invalid syntax? - http://i.imgur.com/aNkrCl0.png

**Comment by wb9688** [comment]
@Legitgamer264: What Python version are you using? It works fine in the
latest version of Python 2.7…

**Comment by Legitgamer264** [comment]
@wb9688 I am using 3.5.2
Should I try python 2.7?

**Comment by wb9688** [comment]
@Legitgamer264: Nope, it should work if you replace the space after every
`print` statement with a `(` and add a `)` at the end of that line…

**Comment by wb9688** [comment]
@Legitgamer264: Btw `raw_input` is renamed to `input` in Python 3, so you
should also change that…

**Comment by Legitgamer264** [comment]
@wb9688 I just uninstalled python 3.5.2 then reinstalled the requests module. It worked now

**Comment by quasimoju** [comment]
can't open your osx version. just getting the error that it can't open.

**Comment by tbraz** [comment]
Since the beginning of 2017 that I haven't been able to play anything with the app. Getting the 429 HTTP error.
How come I keep getting above my API daily limit, even when I don't use the app for more than 24h?

**Comment by weblancaster** [comment]
@tbraz the limit we have is based on all users using the app not individually. unfortunately.

**Comment by tbraz** [comment]
@weblancaster that makes more sense, but it really is unfortunate, as you said. Anyway, thanks for the feedback. All that is left is to wait for Soundcloud to update their usage policy :(

**Comment by lbru** [comment]
@wb9688's python script worked like a charm for me and it apparently persists between launches. It is probably against SC's TOS, but implementing that in Soundnode would fix the issue.

**Comment by jacklaag** [comment]
@wb9688 cool ! your fix works :)
Seems like in 2017 Soundnode got a bit too polular for the Client ID limit..

**Comment by weblancaster** [comment]
@wb9688 can you open a PR for Soundnode that contain that workaround you did please? I want to take a look and maybe start using it.

**Comment by wb9688** [comment]
@weblancaster: I'll do that when I've time, but I've to port it to the new Electron version first…

**Comment by fozzle** [comment]
@wb9688 clever workaround. I just wanted to mention here for anyone reading that if you don't want to use the modified sound node he provided, the way to use the client ID and access token that you get from that python script is as follows:

1.) Open Soundnode
2.) Open developer tools, go to tab `Resources`, under local storage open `file://`
4.) Add the Access Token under the key `scAccessToken`
5.) Add the Client ID under the key `scClientId`
6.) Restart Soundnode

**Comment by eigenstil** [comment]
@fozzle 
> Open developer tools

How do I open the developer tools?

**Comment by wb9688** [comment]
@eigenstil: In the NW.js version of Soundnode, you can press Ctrl+/ to open
the devtools…

**Comment by eigenstil** [comment]
@wb9688 I have a german keyboard layout, thus I have to press "Shift + 7" to insert a "/". So I tried to type "Ctrl + Shift + 7", but this doesn't work either.

What am I doing wrong?

**Comment by pendis** [comment]
@fozzle i tried this but when i started up soundnode again everything was blank (buttons and stuff were there but no songs, icons, etc)

**Comment by themanjotsingh** [comment]
@pendis using the modified version?

**Comment by vasumalhotra** [comment]
@themanjotsingh I tried opening the modified version but it won't open

**Comment by vasumalhotra** [comment]
@pendis I have the exact same issue

**Comment by wb9688** [comment]
@vasumalhotra: On what platform? I only tested my modified version on Linux
and Windows, where it works, because I've no macOS, but someone reported to
me that it doesn't work on macOS…

**Comment by wb9688** [comment]
@vasumalhotra: Ah, but what do you mean?

**Comment by vasumalhotra** [comment]
@wb9688 Or do you know what I might be able to do to get Soundnode to play?

**Comment by vasumalhotra** [comment]
@wb9688 like from icyarrow.com

**Comment by vasumalhotra** [comment]
@wb9688 honestly I'm really new to programming, so I'm sorry but I'm not really sure, is there anyway you could give me some step-by-steps?

**Comment by wb9688** [comment]
@vasumalhotra: Yes:

1. Copy that script into a file called `sc.py` in your user's folder
2. Open a terminal and type in `python sc.py`
3. Type in your username and password
4. Open the regular Soundnode (pre-Electron) and open the devtools with
Ctrl+/ or somewhere in the top menu in macOS
5. Type `localStorage.scAccessToken = ` and paste the access token the
Python script gave you after it
6. Type `localStorage.scClientId = ` and paste the client ID the Python
script gave you after it
7. Restary Soundnode and it should work

**Comment by vasumalhotra** [comment]
@wb9688 I'm trying to find the sc.py file but I'm having trouble locating it, should I just do a search through Finder?

**Comment by wb9688** [comment]
@vasumalhotra: There's no `sc.py` file, you should make a `sc.py` (plain
text) file, where you should paste in the Python code…

**Comment by vasumalhotra** [comment]
@wb9688 Oh gotcha

**Comment by ashlacey** [comment]
Hi @wb9688,

Thanks for the python script; I've tried running it however but it doesn't return any AcessToken or ClientID. This is what I get after entering my username and password into Terminal. Noob to programming as well here! Should I be logged into the site and or have soundnode running whilst doing this? 

Traceback (most recent call last):
  File "sc.py", line 11, in <module>
    access_token = requests.post('https://sign-in.soundcloud.com/sign-in/password?client_id={}'.format(client_id), data=json.dumps({'client_id': client_id, 'scope': 'fast-connect non-expiring purchase signup upload', 'recaptcha_response': '', 'credentials': {'identifier': username, 'password': password}}), headers={'Content-type': 'application/json'}).json()['session']['access_token']
KeyError: 'session'

**Comment by wb9688** [comment]
@ashlacey: That means one of the following:
1. You've mistyped your username and/or password
2. You don't have password authentication enabled in your SoundCloud account
3. You need to solve a CAPTCHA, which isn't yet supported by my Python
script, so you need to log in at SoundCloud.com, log out, wait a minute and
retry running my Python script
4. There's something wrong at SoundCloud

**Comment by smith153** [comment]
Installed soundnode yesterday and it worked. Today it wouldn't play. Used the script trick posted by @wb9688   thanks

**Comment by JHelar** [comment]
@eigenstil 
You can access the dev tools in v0.6.5, just change your local input language for your keyboard to en-us layout. Then press ctrl + - (the "-" key on a EU layout keyboard corresponds to US's "/" ).

The script solution seems to be working! :)

**Comment by eigenstil** [comment]
@JHelar @wb9688 you guys are awesome!

**Comment by jarrz** [comment]
@wb9688 sorry for asking a stupid question but I've been trying to get this fix to work for the past hour, could you tell me how to run that script in windows? I'm not a programmer at all so I have no idea what I'm doing but I did find the developer options. When I put my sc username as ClientID it didn't work anymore and I had to log in again.

**Comment by eigenstil** [comment]
@jarrz I downloaded and installed the "new" version, that @wb9688 linked in his comment (https://github.com/Soundnode/soundnode-app/issues/802#issuecomment-261705999). 

The download link is here: https://wb9688.stackstorage.com/s/7NOaUKaMP9GPXCB

I unzipped this file on my windows and launched the exe-file. After that, everything works like a charm :-)

**Comment by jarrz** [comment]
@eigenstil thanks, but I tried that and it doesn't authenticate my account, even when I have sc open with my account logged in

**Comment by eigenstil** [comment]
@jarrz Ok, sorry - I don't have any further knowledge about this.

**Comment by MrJozroz** [comment]
@wb9688  Okay, a two issues. I can't seem to login, I must have mistyped my login creds, resulting in it endlessly attempting to authenticate. Deleting the folder and redownloading it still starts it up with the authentication loop. How do I reset it?

@eigenstil @JHelar Regarding accessing devtools, I can't seem to access that keyboard shortcut, even when I change to US keyboard ( "?" and "/" are bound to the same key, with shift modifying it to "?" and pulling up the shortcut menu). I browsed through different keyboards and found that the Belgian comma keyboard has "/" free standing, but it doesn't respond to that key any way. Any clues?

**Comment by eigenstil** [comment]
@MrJozroz on my windows 10 it worked. you have to press the "-" hyphen symbol (on the german keyboard located directly next to the right Shift key): 
![skizze](https://cloud.githubusercontent.com/assets/13272281/22705790/378a4858-ed6d-11e6-8d4b-0903eb2e662f.png)

**Comment by MrJozroz** [comment]
@eigenstil Aha, "Ctrl" rather than "Shift". Gotcha, thanks!

**Comment by ce0311** [comment]
Is there any elegant way to use my own API key without manually generating my access token?
I got my own Soundcloud developer ID, added it to main.js and copied the source code from sc-redirect.herokuapp.com to my own server and changed the client ID and secret there too.

I'm just wondering where I can point the app to my own server instead of sc-redirect.herokuapp.com. Maybe I'm blind?

Is there any other magic I need to keep in mind for the sc-redirect? It feels like there's more going on than what can be replicated by copy and pasting its HTML source.

**Comment by weblancaster** [comment]
@ce0311 if you have a API key provided by soundcloud you can change the code use your key and that should do it.

**Comment by asessa** [comment]
If anyone have problems with python, I've ported it to bash:

https://gist.github.com/asessa/ee096da2d46a0eeb7d317ef72c3cf610

requirements: bash/zsh, curl 7.40+
tested on macOS 10.12, Debian Wheezy

**Comment by ce0311** [comment]
Any idea why webpack just stopped working for me?

`> Soundnode@0.6.5 webpack:dev /Users/user/GitHub/soundnode-app
> webpack -d --watch --config ./webpack.dev.js

/Users/user/GitHub/soundnode-app/webpack.dev.js:29
    new webpack.optimize.OccurenceOrderPlugin(),
    ^

TypeError: webpack.optimize.OccurenceOrderPlugin is not a constructor
    at Object.<anonymous> (/Users/user/GitHub/soundnode-app/webpack.dev.js:29:5)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at requireConfig (/Users/user/GitHub/soundnode-app/node_modules/webpack/bin/convert-argv.js:96:18)
    at /Users/user/GitHub/soundnode-app/node_modules/webpack/bin/convert-argv.js:109:17`

node v7.6.0
npm  v4.1.2

**Comment by wb9688** [comment]
@assessa: Your Bash script doesn't do the same thing as my Python script…

**Comment by jakejarrett** [comment]
@ce0311 Which webpack version are you using?

**Comment by ce0311** [comment]
@jakejarrett 2.2.1

**Comment by jakejarrett** [comment]
Webpack 2 requires a new config, try installing webpack 1 for now.
```
npm i --save webpack@1.12.0
```

Or, you can revert to the commit before the dependencies were updated
```
git reset --hard 6c0b5459cbca5649236e6af8c2e28469f42cad70 && rm -rf node_modules/ && npm i
```

**Comment by ce0311** [comment]
That fixed it, got the authentication to work now and it plays tracks. Thanks! Some small things are amiss though: No 'quit Soundnode' option in the menu and playing tracks seems to be hit and miss, sometimes with a long delay or failing. Anyone seen this before or any small detail I missed before compiling?

**Comment by weblancaster** [comment]
@ce0311 the long delay is because I have introduced a check before start playing any tracks, meaning before start playing any tracks request the track endpoint and make sure I can play so that depends on the Soundcloud API response time.

**Comment by ce0311** [comment]
@weblancaster Ah, gotcha. If you load long mixes (1h+) it sometimes takes up to 20 seconds depending on your connection. Scrubbing works great now but not sure if it's worth the tradeoff when quickly flicking through the stream to check out new tracks.

**Comment by weblancaster** [comment]
@ce0311 yes a tradeoff, if anyone have a better idea how we can implement this check without blocking the player would be great.

**Comment by luckydays** [comment]
Hey guys, anybody has the same issue with the @wb9688 wonderful 'Soundnode-WBAuth' app?
I've got this thing rolling all the time and the app doesn't start:
![](https://d1ro8r1rbfn3jf.cloudfront.net/ms_120395/GqiOmmQdFralKmjJjptF573LqdtBf5/Soundnode%2B2017-03-03%2B13.14.45.png?Expires=1488629741&Signature=TtouN74lW3PUBh96UvII8qoWfbe36NeQ7NUJ~s7pxDExBGaBQeMTmjPyo8VjJn-3Vs1ZLKS-fatMdajgibp~~qhhz8SBhysu1J2gfQpkMabuPKxwGxX8ySzLLArZiuZAVok9sMkM-57HeqU1q3W8f0XVpGN~Sb4NHUGEnolCzmwnExdJN8742oyRyRUMSNFaKJP3xRZIJ0q6iU6E3-Gr2~7RzHnSDBknM5R8AMaRb-HYbaPmQn5OF9EzKpgItce-PPrJz~~5Ck45yo0k~ZMIa1sIGNBJ1KnOWgsBQWJv5ltMxQY3iHQ3Em20oYwcsaDm2p9Eybd~49XohA7KEYMfIg__&Key-Pair-Id=APKAJHEJJBIZWFB73RSA)

I've already re-logged in inside my Soundcloud account on the site, but didn't help.

**Comment by wb9688** [comment]
@luckydays: What OS are you using? I've only tested it on Linux (Debian
Unstable) and Windows (10). Also try removing ~/.config/Soundnode-WBAuth
(Linux) or %appdata%/Soundnode-WBAuth iirc (Windows).

**Comment by luckydays** [comment]
Hi @wb9688 

Thanks for coming back so quick!
I'm on Windows 10.
So, I've deleted the 'Soundnode-WBAuth' folder from '%appdata%/Local' path.
Started the app again → it asked for login and password anew → after filling the fields the app doesn't load still.

Could you suggest something else?

And this issue started just from yesterday.

Thanks!

**Comment by wb9688** [comment]
@luckydays: Are you sure username/password authentication (and not only
Facebook or Google authentication) is enabled in your SoundCloud? If so,
are you sure you've typed your username and password correctly? If so, does
your username or password contain any weird characters? If not, does it
work when you log in with email address instead of username? If not, does
my Python script work?

**Comment by luckydays** [comment]
@wb9688 

>Are you sure username/password authentication (and not only Facebook or Google authentication) is enabled in your SoundCloud?

Sure. I've never connected neither Facebook nor Google account authentication. Only email.

>If so, are you sure you've typed your username and password correctly?

Yep. Tried multiple times.

>If so, does your username or password contain any weird characters?

No. Regular letter and digits, without special characters.

>If not, does it work when you log in with email address instead of username?

Actually I'm trying to login exactly with my account's email. Also tried now with the username - didn't work as well.

>If not, does my Python script work?

I hope it does. How could one check it?
(sorry, I'm not familiar with Python).

**Comment by wb9688** [comment]
@luckydays: Weird…
1. Install Python 2 (not 3!)
2. Copy-paste the script into a plain text file with .py as extension
3. Open your Terminal (or Command Prompt on Windows) and type `python
path/to/script.py`
4. Type your username and password
5. Remember the Client ID and the Access Token it prints
6. Open a normal (nw.js, not Electron) Soundnode and press Ctrl+/
7. In the console, type `localStorage.scAccessToken = "acces-token"` and
`localStorage.scClientId = "client_id"`
8. Restart Soundnode

**Comment by luckydays** [comment]
@wb9688 Could you please clarify couple things from the list?

2. You mean [this script](https://github.com/Soundnode/soundnode-app/issues/802#issuecomment-261705999), right?
3. I've typed the path to the Python27 folder and nothing happened.
6. Could you clarify this, please? Where could I find the normal 'nw.js' file?

**Comment by wb9688** [comment]
@luckydays:
1. Yep.
2. Type `python ` and then the path to the Python script you've saved
somewhere, for example `python C:\Users\wb9688\sc.py`.
3. I mean just the regular Soundnode, downloaded from soundnodeapp.com.
Recently Soundnode migrated from nw.js to Electron, but I haven't looked
into the new Electron version of Soundnode yet, so you need the old nw.js
version of Soundnode.

**Comment by luckydays** [comment]
@wb9688 

After typing 'python ...' I've got such an error:

```
C:\Users\Paul>python C:\Users\Paul\Desktop\script.py
Traceback (most recent call last):
  File "C:\Users\Paul\Desktop\script.py", line 1, in <module>
    client_id = re.compile(',client_id:"(.*?)"').search(requests.get(re.compile('src="https://a-v2.sndcdn.com/assets/app-.*.js"').search(requests.get('https://soundcloud.com/').text).group()[5:-1]).text).group()[12:-1]
NameError: name 're' is not defined
```

**Comment by wb9688** [comment]
@luckydays: Did you copy it fully? It says you haven't copied at least the
`import re` line…

**Comment by luckydays** [comment]
@wb9688 Yes, you're right. In the above case I've deleted first 4 lines (with 'import' in the line's start) because I got another kind of error:

```
C:\Users\Paul>python C:\Users\Paul\Desktop\script.py
Traceback (most recent call last):
  File "C:\Users\Paul\Desktop\script.py", line 4, in <module>
    import requests
ImportError: No module named requests
```

and exactly this one appears now, after I've fully pasted [your python code from here](https://github.com/Soundnode/soundnode-app/issues/802#issuecomment-261705999).

P.S.: thanks a lot for your help and assist with this thing!

**Comment by wb9688** [comment]
@luckydays: Oh, you also have to install Requests with `pip install
requests`.

**Comment by luckydays** [comment]
@wb9688 OK, yeah 'requests' installation helped 😄 
After running 'python C:\Users\ ... script.py' I've got this one:

```
Traceback (most recent call last):
  File "C:\Users\Paul\Desktop\script.py", line 11, in <module>
    access_token = requests.post('https://sign-in.soundcloud.com/sign-in/password?client_id={}'.format(client_id), data=json.dumps({'client_id': client_id, 'scope': 'fast-connect non-expiring purchase signup upload', 'recaptcha_response': '', 'credentials': {'identifier': username, 'password': password}}), headers={'Content-type': 'application/json'}).json()['session']['access_token']
KeyError: 'session'
```

And on this step you told to "Remember the Client ID and the Access Token it prints" - but have no idea where to find it now.

**Comment by wb9688** [comment]
@luckydays: Something is going wrong, remove the lines starting with
`print`, replace `access_token =` with `print` and remove
`['session']['access_token']` at the end of that line, then post it's
output.

**Comment by luckydays** [comment]
The output after these actions is:

```  File "C:\Users\Paul\Desktop\script.py", line 11
    print = requests.post('https://sign-in.soundcloud.com/sign-in/password?client_id={}'.format(client_id), data=json.dumps({'client_id': client_id, 'scope': 'fast-connect non-expiring purchase signup upload', 'recaptcha_response': '', 'credentials': {'identifier': username, 'password': password}}), headers={'Content-type': 'application/json'}).json()
          ^
SyntaxError: invalid syntax
```

**Comment by wb9688** [comment]
@luckydays: You forgot to remove the `=`.

**Comment by luckydays** [comment]
@wb9688 Oops. Sorry for that.

OK, so this time it asked again for the login and password.
After entering those I've got

```{u'error': u'captcha_required'}
```

**Comment by wb9688** [comment]
@luckydays: A CAPTCHA is required, but my script doesn't support that yet.
All you can do is log in on SoundCloud.com and try again with the original
version of my script. If it still doesn't work, you should wait a hour or
so. Btw if you're using a VPN/proxy/whatever, you should disable it, since
SoundCloud requires a CAPTCHA when it thinks you're suspicious.

**Comment by luckydays** [comment]
@wb9688 And is there any other way how one could get client ID and access token - to put them through the original app developer tools'?

**Comment by luckydays** [comment]
@wb9688 Suddenly your Soundnode WB app started to work again 😉

**Comment by Kaizerus** [comment]
Idk what's wrong, I don't code, I'm just on github for the fun stuff, and I'm going to start coding soon, maybe a 1-2 years, or possibly less, all I get is **_Failed to load resource: the server responded with a status of 429_** (Unknown) is there a fix for this? Cause I can't play any music **_AT ALL_**, every track I click, the console shows that as an answer.

**Comment by manjusr81** [comment]
I downloaded your linux64 package.   But how do u install it.   I too have the same problem

**Comment by manjusr81** [comment]
I am using ubuntu 16.04

**Comment by merksam** [comment]
Hello guys, your app is awesome, but I have an error from **Failed to load resource: the server responded with a status of 429 (Unknown)** too :(

**Comment by weblancaster** [comment]
@merksam we have a limit of 15k plays imposed by Soundcloud, for that reason you weren't able to stream tracks.

**Comment by ALoGeNo** [comment]
Hello all, I'm using the "new" version, that @wb9688 linked in his comment #802, I can login perfectly now, I can add favorites, add to playlist, also sharing and copy the song link, but I cant repost any song, I'm getting an "Oh no, something went wrong!" when I click on repost.

Thanks to all for this great software, Cheers!.

**Comment by ninja85a** [comment]
hey ive got the latest version and im getting the same issues everyone else is getting ive tried what other people at the start have tried to fix it but it doest work any ideas?

**Comment by wb9688** [comment]
@ninja85a: Have you read my post? It contains a Python script you can use
to get a token without rate limit and a link to a modified version of
Soundnode that does that automatically. It's just a temporary workaround
until I've time to work further on it.

**Comment by ninja85a** [comment]
@wb9688 ah sorry didnt see so where do i put the python file in the folder with the program?

**Comment by wb9688** [comment]
@ninja85a: For instructions on using my manual workaround, read eg. my
comment starting with "uckydays: Weird…".

**Comment by ninja85a** [comment]
@wb9688 well ive installed python 2 but when i do python in the cmd it doesnt work it says 'python' is not recognized as an internal or external command,
operable program or batch file.

**Comment by wb9688** [comment]
@ninja85a: Then Python isn't in your path, so you could:
1. Add Python to your path
2. Type the full path to Python

**Comment by ninja85a** [comment]
@wb9688 huh ive followed a guide on how to add it to the path and its already added

**Comment by wb9688** [comment]
@ninja85a: That's weird. Then try my second option, eg. typing something
like `C:\Python27\python.exe C:\path\to\script.py` on Windows.

**Comment by ninja85a** [comment]
okay that got it sort of working this is the error i got
Traceback (most recent call last):
  File "C:\Users\ninja\Downloads\Soundnode\soundcloud.py", line 4, in <module>
    import requests
ImportError: No module named requests

**Comment by wb9688** [comment]
@ninja85a: Then you haven't installed the `requests` Python package.
There's probably some `pip.exe` in your Python folder, which could be used
to install it, by running `C:\path\to\pip.exe install requests`.

**Comment by ninja85a** [comment]
@wb9688 nice got it working now hopefully all the other bits will work cheers

**Comment by ninja85a** [comment]
@wb9688 it worked!!! cheers 😃

**Comment by weblancaster** [comment]
I have made the oauth open for those that have the Soundcloud `id` and `secret` so now you are able to host yourself and not have problems with limit reached. https://github.com/Soundnode/sc-oauth

**Comment by piotryordanov** [comment]
@wb9688 Your solution works perfectly for me. Wondering why that isn't in the Readme somewhere! :)

**Comment by matiaslabra** [comment]
@wb9688 thanks!! it worked like a charm

**Comment by Skaifer** [comment]
Hello! Is it working on windows 10? Just when I try to use it nothing happen, only after rewind it throws that error.
![screenshot_1](https://cloud.githubusercontent.com/assets/28566054/25852865/77608c14-34d3-11e7-9cc7-582ca7a50e9a.png)

**Comment by rodrigoblz1** [comment]
@wb9688 great, it's working for me, just two things I had to deal with:
1. (OS X Sierra) Permission to the .app after have moved it to the Applications Folder. (It was giving me an error message)
2. I mistyped my password so I had to go to resources > local storage and delete it, otherwise it was giving me some 'undefined token' error on console and never authenticating.

**Comment by weblancaster** [comment]
@Skaifer you are probably facing the same issue majority of us do.. Hitting the rate limit Soundclooud imposes in third party apps.

To by pass that you can use what @wb9688 mentioned here or ask soundcloud to [register a new](https://developers.soundcloud.com/docs/api/guide) third party app and you can use the key in Soundnode.

**Comment by Skaifer** [comment]
@weblancaster Thank you for response! Can't guess what takes it to register an app a month. Maybe I'll try out python script later then. Why there is nothing mentioned on homepage? In terms of temporary issue or what? Maybe then there will be less idiots like me making the same question over and over. =)

**Comment by weblancaster** [comment]
@Skaifer they just don't want to register apps anymore, only a handful.

I do need to add to the homepage in the website but we do have included in the homepage on Github.

`be aware...` section

![screen shot 2017-05-15 at 8 40 12 am](https://cloud.githubusercontent.com/assets/549394/26065967/5346d542-394a-11e7-970c-67017cbc1a21.png)

**Comment by Skaifer** [comment]
@weblancaster Ok, thanks.

**Comment by usersxm** [comment]
Not a single song is being played. I don't know why, I log in (using google authentication) press play on any song, the notification is displayed "you're playing xxx" but no sound and no controls over the track, to sum up songs are not being played.

**Comment by weblancaster** [comment]
@usersxm just read two comments above (or this entire tread) and you will understand why that is happening to you.

**Comment by usersxm** [comment]
@weblancaster I've read that but I thought that it was another issue, because if it was an API problem I thought that there might be a warning notification of limit reached

**Comment by Nixontira** [comment]
@weblancaster 

> I have made the oauth open for those that have the Soundcloud id and secret so now you are able to host yourself and not have problems with limit reached. https://github.com/Soundnode/sc-oauth

hey !
I just read this and im trying to make this work but I have no clue how.. I downloaded the stuff from GitHub, but what do I have to do with those files? I tried to run the Procfile but then terminal gives me an error. Im using the newest Sierra version

**Comment by Skaifer** [comment]
Well, they have gave me the key. But where to enter it? - Windows 10

**Comment by weblancaster** [comment]
@Nixontira you will need to run the server somewhere where you can have a constant domain.

I have added a quick guide on how to do. https://github.com/Soundnode/sc-oauth/blob/master/README.md

**Comment by weblancaster** [comment]
@usersxm I have added a notification but haven't shipped yet. sorry.

**Comment by chrismessina** [comment]
@weblancaster I wonder if it'd make sense to follow the model of [Spotificious](https://github.com/citelao/Spotify-for-Alfred), a Spotify controller for Spotify built for Alfred. It [runs a local PHP server](http://ben.stolovitz.com/Spotify-for-Alfred/download/) in which you inject your Spotify developer token to get around the rate limiting problem.

It just seems like folks are going to continue to struggle w/ what to do w/ their auth tokens until something like this (i.e. a step-by-step guided solution) is offered. WDYT?

**Comment by weblancaster** [comment]
@chrismessina correct, I have made https://github.com/Soundnode/sc-oauth available so people with more experience in development can run locally, But Soundcloud API requires to register under a domain.

**Comment by wb9688** [comment]
@weblancaster: Could you please respond to my questions in #997? Thanks!

**Comment by weblancaster** [comment]
@wb9688 I don't see a question there.

**Comment by wb9688** [comment]
@weblancaster: Aren't those questions?

>@weblancaster: Couldn't you add a comment template stating something like
the following?
>
>Please check if your issue is already reported. If you can't play tracks,
please take a look at #802.
>
>---
>
>@weblancaster: Since I've no experience with AngularJS (1), could you
explain (or link to) how I can add a page (for logging in)?

**Comment by pjf** [comment]
Process I used for getting this working under the new electron app:

1. Clone the git repo and change into the directory.
1. `npm i` (to install dependencies)
1. `npm start` to run soundnode. This will create the config file you'll need to edit.
1. Exit Soundnode.
1. Use @wb9688's [excellent code](https://github.com/Soundnode/soundnode-app/issues/802#issuecomment-261705999).
1. Edit the `app/public/js/system/userConfig.json` file, replacing the `accessToken` and `clientId` details with those from wb9688's script above.
1. Start the app as normal (`npm start`) and enjoy the music!

Many thanks to everyone for both giving me a sweet desktop soundcloud app, and the technical hints to get it running despite rate-limits!

**Comment by jakejarrett** [comment]
@weblancaster It looks like the question is minimised by github (The `...` at the bottom of the comment)

**Comment by Starima** [comment]
@pjf May I ask if you'd be so kind and explain this in a way more suitable for some guy with no actual clue like me? I'm trying to fix this since 1h and I don't make any kind of progress. It's pretty frustrating..
Thanks in advance. :/

**Comment by weblancaster** [comment]
Everyone take a look at this issue and maybe we can have a solution soon #999.

**Comment by br4nnigan** [comment]
@wb9688 why do media buttons not work with your version? Can that be fixed?

**Comment by wb9688** [comment]
@br4nnigan: I don't know, since I haven't really looked at SoundNode for a
long time and I don't have media keys. What OS are you using though? If
it's Windows or Linux, I can maybe take a look at it.

**Comment by br4nnigan** [comment]
Yeah it's Windows and I have a simple MS Wired keyboard. Would be dope because this project still doesn't seem to have a solution to the API limit.

**Comment by wb9688** [comment]
@br4nnigan: I've looked at it, but I couldn't fix it, because I haven't
found a way to emulate media key presses on Windows, sorry.

**Comment by br4nnigan** [comment]
@wb9688 god bless you my friend

**Comment by brodycas3** [comment]
@wb9688 Your version now seems to be stuck, https://i.imgur.com/TwtZxdx.png

**Comment by wb9688** [comment]
@brodycas3: Try deleting the Soundnode-WBAuth folder in %appdata%\Local or
~/.config. If that doesn't work, SoundCloud probably thinks you're a bot
and requires a CAPTCHA, which isn't supported by my version. Sometimes it
helps to log into SoundCloud.com then.

**Comment by DamSenViet** [comment]
I still can't get in. I usually log in through google.

Here's what I've attempted:

My actual username (it's user-BunchOfRandomNumbersGeneratedBySoundcloud)

My gmail

Both seem to stall the authentication. I've also tried logging into soundcloud.com but doesn't seem like its budging.

**Comment by Nixontira** [comment]
Could someone explain wehre i find The localstorage file under macOS ?

**Comment by star114** [comment]
@Nixontira 
[Here's instruction for macOS](https://github.com/Soundnode/soundnode-app/issues/802#issuecomment-259316380)

**Comment by 1syn** [comment]
@wb9688 I tried running your application that you uploaded to STACK on OSX ElCapitan 10.11.6, and was given the prompt "The application "Soundnode-WBAuth" can't be opened." The prompt is not helpful to me and probably not helpful to you. I'm not good with programming (yet =] ), but I see that you referenced client_id in the python script. Maybe the app isn't working because SoundCloud stopped giving IDs out or maybe its just an id10t error, but if you could shed some light on why I'm getting this error I would be greatly appreciative.

**Comment by IntrovertBoy** [comment]
Hi. The same problem that was posted here happened to me. I just installed this app, ran it but songs didn't play and it only gave me an error. I read your solutions above but didn't understand anything that I have to do to get it working ( I tried clearing Local Storage and restarting didn't help). Can someone please explain to me what I have to do?

**Comment by gnuletik** [comment]
As Soundcloud doesn't allow the creation of new API keys (see https://soundcloud.com/you/apps/new ), @wb9688's [workaround](https://github.com/Soundnode/soundnode-app/issues/802#issuecomment-261705999) is not working anymore right ?

So, once the limit is reached (error `the server responded with a status of 429 (Unknown)` in logs), there is nothing we can do about it right ? That's too bad

**Comment by dblythy** [comment]
I hope someone can build this, but here's how to avoid API keys altogether:

I created an embed player with the URL requested to be played. Set the player to hidden, and used my own media controls to control the hidden embed player. So I used my own facing to control a hidden SoundCloud embed player.

You need the player https://w.soundcloud.com/player/?url= and the track permalink_url.

**Comment by RepComm** [comment]
So do we have any brain storms about where to go next? Obviously we're going to loose a lot of users if it isn't convenient, let alone not working. SoundCloud looks like they're trying to get rid of external players, so maybe it's time for a switching of platforms.. It would be interesting to get on board with a custom platform altogether. I disagree with a lot of sound clouds policy on a personal level, so I'm game with trying some new things..

**Comment by smith153** [comment]
You do not need an "API Key". You need your "device id". How you get that, got to soundcloud.com and logout (if you are logged in). Then open your web browser's dev console (ctrl+shift+c on firefox). Go to the networking tab. Now log in as usual and when you hit enter you will see a POST request sent. Examine that request and in the data that is returned is you device id, it will be 32 characters long. Copy that into soundnode and you are good to go.

**Comment by HWideman** [comment]
To add my "device id" do I need to compile my own version of Soundnode? If so, where would one add the device id?

**Comment by victorgawk** [comment]
I have cleaned my local storage and logout from web browser. Then after I relaunched the app it worked!

**Comment by mdxprograms** [comment]
This is still an issue btw. I would love to use this over something like vox, but even when I clear local storage and restart app I get the same thing. I've signed out of soundcloud in my browser as well.

**Comment by npblubb** [comment]
@jakejarrett not having an API key is the root issue here ...

**Comment by felipevalois** [comment]
> You do not need an "API Key". You need your "device id". How you get that, got to soundcloud.com and logout (if you are logged in). Then open your web browser's dev console (ctrl+shift+c on firefox). Go to the networking tab. Now log in as usual and when you hit enter you will see a POST request sent. Examine that request and in the data that is returned is you device id, it will be 32 characters long. Copy that into soundnode and you are good to go.

Does this work^? How does one achieve that

**Comment by db4z** [comment]
Any solution to this problem?

**Comment by SW1FT** [comment]
> 
> 
> Any solution to this problem?

Use [SoundCleod](https://github.com/salomvary/soundcleod) as an alternative to Soundnode.

**Comment by db4z** [comment]
> > Any solution to this problem?
> 
> Use [SoundCleod](https://github.com/salomvary/soundcleod) as an alternative to Soundnode.

Thanks for letting me know about this one, but I'm using Linux Ubuntu.

**Comment by czerwinski76** [comment]
It's not à 15K issue. Use Auryo !😀.
https://auryo.com/

**Comment by db4z** [comment]
> It's not à 15K issue. Use Auryo !.
> https://auryo.com/

This one works well on Linux, cheers mate!

**Comment by JohnAshraf37** [comment]
> 
> 
> You do not need an "API Key". You need your "device id". How you get that, got to soundcloud.com and logout (if you are logged in). Then open your web browser's dev console (ctrl+shift+c on firefox). Go to the networking tab. Now log in as usual and when you hit enter you will see a POST request sent. Examine that request and in the data that is returned is you device id, it will be 32 characters long. Copy that into soundnode and you are good to go.

If anyone still wondering this method works
